### PR TITLE
added implement Hibernate in User, Bucket, Item, Role

### PR DESCRIPTION
**Pull request by TarasHryts**

### CODE REVIEW

File: src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    @Inject
+    private static ItemDao itemDao;
+    @Inject
+    private static UserDao userDao;
+
+    @Override
+    public Optional<Bucket> create(Bucket bucket) {
+        Long bucketId = null;
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession();) {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java]: try to use block finally in methods create, update and delete

File: src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+        return Optional.of(bucket);
+    }
+
+    @Override
+    public void addBucketForUser(Long userId, Long bucketId) {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java:108]: do we need this method?

File: src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            Bucket bucket = get(bucketId).get();
+            List<Item> itemList = bucket.getItems();
+            itemList.add(itemDao.get(itemId).get());
+            bucket.setItems(itemList);
+            update(bucket);
+        }
+    }
+
+    @Override
+    public List<Item> getItemForBucket(Long bucketId) {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java]:
```suggestion
    public List<Item> getItemsFromBucket(Long bucketId) {
```

File: src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java (hunk 1 above)
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java]:
```suggestion
        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
in all try with resources

File: src/main/java/mate/academy/shop/dao/hibernate/RoleDaoHibernateImpl.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+        }
+    }
+
+    @Override
+    public Set<Role> getAllRoleForUser(Long userId) {
+
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/dao/hibernate/RoleDaoHibernateImpl.java:101]: remove one empty line

**Review by Sofasmile — COMMENTED** [review]
good, make small fixes

File: src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java (hunk 1 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java]: done

File: src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java (hunk 2 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java:108]: I will delete all of it, when I will clean the project

File: src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java (hunk 3 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java]: done

File: src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java (hunk 1 above)
Comment (TarasHryts) [reply src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java]: done

File: src/main/java/mate/academy/shop/dao/hibernate/UserDaoHibernateImpl.java [hunk 5]
```diff
@@ -7,71 +7,101 @@
...
 
     @Override
     public List<User> getAllUsers() {
-        try (Session session = HibernateUtil.getSessionFactory().openSession();) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
             return session.createQuery("FROM User").list();
+        } catch (Exception e) {
+            logger.error("Can't get users");
         }
+        return null;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/shop/dao/hibernate/UserDaoHibernateImpl.java:104]: return Collections.emptyList()

**Review by Sofasmile — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -1 +1,38 @@
-# store
\ No newline at end of file
+
```
Comment (Sofasmile) [inline README.md:1]: add name of the project: Internet shop

**Review by Sofasmile — APPROVED** [review]
make small fixes)

**Pull request merged**